src/main/java/nl/igorski/lib/audio/MWEngine.java

```diff
@@ -200,11 +200,34 @@ public void setBouncing( boolean value, String outputDirectory, int maxRecordBuf
     }
 
     /**
-     * records the live output of the engine
+     * Records the audio coming in from the Android device input.
+     * Requires RECORD_DEVICE_INPUT to be enabled in global.h as well
+     * as the appropriate permissions in the AndroidManifest.
      *
-     * this keeps recording until setRecordingState is invoked with value false
-     * given outputDirectory will contain several .WAV files each at the buffer
-     * length returned by the "calculateMaxBuffers"-method
+     * In order to record the input directly to device storage, @see
+     * setRecordFromDeviceInputState()
+     *
+     * @param record {boolean}
+     */
+    public void recordInput( boolean record )
+    {
```

Comment:
New function to monitor device audio input added. Note that the remaining two record functions have kept their name and signature for backwards compatibility, but now have an extensive documentation of their differing use cases within their Javadocs.